package sudoku;

option cc_generic_services = true;

message SudokuRequest {
   required string checkerboard = 1; 
}

message SudokuResponse {
  optional bool solved = 1 [ default = false ];
  optional string checkerboard = 2;
}

message GcdRequest{
  optional int64 val1=1;
  optional int64 val2=2;
}

message GcdResponse {
  optional int64 res=1;
}

service SudokuService { 
  rpc Solve(SudokuRequest) returns (SudokuResponse); 
  rpc Gcd(GcdRequest) returns(GcdResponse);
}
